/*
 * Copyright (c) 2020-2020, Exactpro (Exactpro Systems Limited)
 * www.exactpro.com
 * Build Software to Test Software
 *
 * All rights reserved.
 * This is unpublished, licensed software, confidential and proprietary 
 * information which is the property of Exactpro Systems Limited or its licensors.
 */

syntax = "proto3";
package th2;

import "infra.proto";
import "rhbatch.proto";
import "google/protobuf/timestamp.proto";

option java_multiple_files = true;
option java_package = "com.exactpro.th2.act.grpc";

message BaseRequest {
    RhSessionID sessionID = 1;
    EventID parentEventId = 2;
}

message ActResponse {

    enum ExecutionStatus {
        SUCCESS = 0;
        ACT_ERROR = 1;
        HAND_ERROR = 2;
        COMPILE_ERROR = 3;
        EXECUTION_ERROR = 4;
    }

    ExecutionStatus scriptStatus = 1;
    string statusInfo = 2;
    string errorInfo = 3;
    map<string, string> data = 4;
    RhSessionID sessionID = 5;
}